doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,224 @@
+| _single-point_ | `point` | [`arma::vec`](../matrices.md) | Single point for classification. |
+| _single-point_ | `prediction` | `size_t&` | `size_t` to store class prediction into. |
+||||
+| _multi-point_ | `data` | [`arma::mat`](../matrices.md) | Set of [column-major](../matrices.md#representing-data-in-mlpack) points for classification. |
+| _multi-point_ | `predictedLabels` | [`arma::Row<size_t>&`](../matrices.md) | Vector of `size_t`s to store class prediction into.  Will be set to length `data.n_cols`. |
+
+***Note:*** different types can be used for `data` and `point` (e.g.
+`arma::fmat`, `arma::sp_mat`, `arma::sp_vec`, etc.).  However, the element type
+that is used should be the same type that was used for training.
+
+### Other Functionality
+
+ * `WeakLearner(size_t i)` returns a `DecisionTreeRegressor&` indicating which weak learner 
+    is being used. 
+
+ * `gb.NumWeakLearners()` returns a `size_t` indicating how many weak learners are being
+    used by the model.
+
+ * `gb.NumClasses()` returns a `size_t` indicating the number of classes the
+   model was trained on.
+
+ * `gb.SetNumWeakLearners(size_t x)` sets the number of weak learners to `x` explicitly.
+
+For complete functionality, the [source
+code](/src/mlpack/methods/grad_boosting/grad_boosting.hpp) can be consulted.
+Each method is fully documented.
+
+### Simple Examples
+
+See also the [simple usage example](#simple-usage-example) for a trivial use of
+`GradBoosting`.
+
+---
+
+Train a gradient boosting model on mixed categorical data and save it:
+
+```c++
+// Load a categorical dataset.
+arma::mat dataset;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/covertype.train.arff.
+mlpack::data::Load("covertype.train.arff", dataset, info, true);
+
+arma::Row<size_t> labels;
+// See https://datasets.mlpack.org/covertype.train.labels.csv.
+mlpack::data::Load("covertype.train.labels.csv", labels, true);
+
+// Create the model object.
+mlpack::GradBoosting gb;
+// Train on the given dataset, specifying the number of weak learners as 5.
+gb.Train(dataset, info, labels, 7 /* classes */, 5 /* number of weak learners */);
+
+// Load categorical test data.
+arma::mat testDataset;
+// See https://datasets.mlpack.org/covertype.test.arff.
+mlpack::data::Load("covertype.test.arff", testDataset, true);
+
+// Predict class of first test point.
+const size_t firstPrediction = gb.Classify(testDataset.col(0));
+std::cout << "Predicted class of first test point is " << firstPrediction << "."
+    << std::endl;
+
+// Save the model to `gb.bin`.
+mlpack::data::Save("gb.bin", "gb", gb);
+```
+
+---
+
+Load a model and print some information about it.
+
+```c++
+mlpack::GradBoosting gb;
+// This call assumes a gb called "gb" has already been saved to `gb.bin`
+// with `data::Save()`.
+mlpack::data::Load("gb.bin", "gb", gb, true);
+
+std::cout << "The number of weak learners being used by the model is "
+<< gb.NumWeakLearners() << "." << std::endl;
```

Comment:
```suggestion
std::cout << "The number of weak learners being used by the model is "
    << gb.NumWeakLearners() << "." << std::endl;
```

Reply:
If you can print more information through some of the other accessors (like maybe a couple things from the first weak learner) that would be nice too.